src/include/storage/projected_row.h

```diff
@@ -210,6 +210,8 @@ class ProjectedRowInitializer {
   template <typename AttrType>
   static ProjectedRowInitializer Create(std::vector<AttrType> real_attr_sizes, const std::vector<uint16_t> &pr_offsets);
 
+  ProjectedRowInitializer() = default;
```

Comment:
I would add a doxygen comment warning about using this constructor, or make it private with catalog and database catalog as friends. We don't wan't people thinking there's other ways to get a pri or coding up their own, aside from our create methods